### DISCUSSION TITLE
Package sjtutex: `cjk-font = fandol' is unavailable in current mode.

### DISCUSSION THREAD

**Discussion by sqinghua** [sjtug/SJTUThesis#1015] [提问] [answered]
请问这个问题是什么意思啊？如何解决？
非常感谢！！
![图片](https://github.com/sjtug/SJTUThesis/assets/71050713/2feaa99d-2160-46a1-ab29-ae9ef07de04e)

**Comment by LogCreative** [comment]
应当使用 XeLaTeX 编译，请详阅 README。

**Reply by sqinghua** [reply]
我之前用的latexmk。
按您说的，用了xelatex后报错
![图片](https://github.com/sjtug/SJTUThesis/assets/71050713/7bd19d25-6e53-4512-981e-8b8e29d8b3dc)
然后我又用latexmk编译了一下，还是报错，另外也没有生成log文件。谢谢！
![图片](https://github.com/sjtug/SJTUThesis/assets/71050713/5ea810cb-a573-4e12-b836-89d66af746f1)

**Reply by LogCreative** [reply]
应当用 latexmk (xelatex)。

https://github.com/sjtug/SJTUThesis/blob/master/README.md#vscode-%E7%94%A8%E6%88%B7

**Reply by sqinghua** [reply]
谢谢！原因知道了！不过按照教程来配置还是失败了，我再去看看别的相关的discussion的解决办法。

**Reply by sqinghua** [reply]
能否帮忙看下错误原因？非常感谢！
按要求配置了json
![图片](https://github.com/sjtug/SJTUThesis/assets/71050713/5f59600f-552c-40c4-a07d-dfeed1ac2ae8)
![图片](https://github.com/sjtug/SJTUThesis/assets/71050713/b74b0e0c-a36b-425f-bafd-0dffab178cdb)


```
=== TeX engine is 'XeTeX'
Latexmk: Errors, so I did not complete making targets
'xelatex -synctex=1 -no-pdf -synctex=1 -interaction=nonstopmode -file-line-error -halt-on-error -recorder  "f:\03_thesis\SJTUThesis-master\main.tex"': time = 0.015
Accumulated processing time = 1.282
Collected error summary (may duplicate other messages):
  xelatex: Command for 'xelatex' gave return code 1
      Refer to 'main.log' for details
Latexmk: Use the -f option to force complete processing,
 unless error was exceeding maximum runs, or warnings treated as errors.
D:\texlive\2021\bin\win32\runscript.tlu:915: command failed with exit code 12:
perl.exe d:\texlive\2021\texmf-dist\scripts\latexmk\latexmk.pl -xelatex -synctex=1 -interaction=nonstopmode -file-line-error -halt-on-error -time f:/03_thesis/SJTUThesis-master/main
```

**Reply by LogCreative** [reply]
建议直接使用默认配置，也可能 tex 文件有其他部分的错误，建议检查一下 main.log，建议先 clean 再编译。使用 latexmk 是为了读取 .latexmkrc 中的配置，引导 TeX 程序正常识别 texmf 文件夹中的依赖文件，也可以尝试直接使用这种配置，即不添加参数：
```json
      {
            "name": "latexmk_rconly",
            "command": "latexmk",
            "args": [
                "%DOC%"
            ],
            "env": {}
        },
```

**Reply by sqinghua** [reply]
感谢解答！

**Reply by sqinghua** [reply, marked as answer]
破案了，是我的texlive2021版本太低，更新到2023就好了！
